src/api.js

```diff
@@ -182,12 +187,22 @@ api.declare({
     throw e;
   }
 
-  // Not all webhook payloads include an e-mail for the user who triggered
-  // an event.
-  let headUser = msg.details['event.head.user.login'];
-  let userDetails = await this.github.users.get(headUser);
+  try {
+    debug(`Trying to authenticate as installation for ${msg.details['event.type']}`);
+    var instGithub = await this.github.getInstallationGithub(msg.installationID);
+  } catch (e) {
+    debug('Error authenticating as installation');
+    throw e;
+  }
+
+  // Not all webhook payloads include an e-mail for the user who triggered an event
+  let headUser = msg.details['event.head.user.id'].toString();
+  let userDetails = await instGithub.users.getById({id: headUser});
+  msg.details['event.head.user.email'] = userDetails.email || 
+    msg.details['event.head.user.login'] + '@users.noreply.github.com';
 
-  msg.details['event.head.user.email'] = userDetails.email || headUser + '@users.noreply.github.com';
+  msg.details['event.head.user.email'] = msg.details['event.head.user.login'] + '@users.noreply.github.com';
```

Comment:
Maybe my eyes are crossing, but it looks like you're seeting the same value twice here..